### ISSUE TITLE
bug(common): kmx struct alignment for more platforms 🙀

### ISSUE THREAD

**Issue by srl295** [keymanapp/keyman#9138]
* in kmx_file.h, this attribute is currently only applied on WASM builds (clang-based), but we need to consider for other compilers as well (clang-other platforms, GCC, VC++)

https://github.com/keymanapp/keyman/pull/9134#issuecomment-1614029004

**Comment by srl295** [comment]
memdup'd in #9139

**Comment by mcdurdin** [comment]
Move 
```
#ifdef __EMSCRIPTEN__
typedef KMX_DWORD __attribute__((aligned(1))) KMX_DWORD_unaligned;
typedef KMX_BOOL  __attribute__((aligned(1))) KMX_BOOL_unaligned;
typedef KMX_WORD  __attribute__((aligned(1))) KMX_WORD_unaligned;
#else
// TODO: consider other platforms
#define KMX_DWORD_unaligned KMX_DWORD
#define KMX_BOOL_unaligned  KMX_BOOL
#define KMX_WORD_unaligned  KMX_WORD
#endif
```

into km_types.h and then resolve the todo in it.

**Comment by mcdurdin** [comment]
See also keyman_core_api_bits.h for cross-platform patterns we should copy for consistency.

**Comment by srl295** [comment]
Fixed in #9977
